server/startup/restful_api/api.js

```diff
@@ -0,0 +1,184 @@
+                  status: "fail",
+                  message: "Record Creation was not succesful"
+                };
+              }
+              return {
+                statusCode: 201,
+                status: "success",
+                data: insertedData
+              };
+            }
+          }
+        },
+
+        put: {
+          // TODO: write the put action() method
+          authRequired,
+          action() {
+            if (!(isPermitted(this.user, "admin")) ||
+              isPermitted(this.user, "owner")) {
+              return {
+                statusCode: 401,
+                status: "fail",
+                message: "you do not have permission to edit this record"
+              };
+            }
+            if (isPermitted(this.user, "admin") ||
+              isPermitted(this.user, "owner")) {
+              const update = Collection.upsert({
+                _id: this.urlParams.id
+              }, {
+                $set: this.bodyParams
+              });
+              if (!update) {
+                return {
+                  statusCode: 404,
+                  status: "fail",
+                  message: "Record does not exist"
+                };
+              }
+              const record = Collection.findOne(this.urlParams.id);
+              return {
+                statusCode: 200,
+                status: "success",
+                data: update, record
+              };
+            }
+          }
+        },
+
+        delete: {
+          // TODO: write the delete action() method
+          authRequired,
+          action() {
+            if (!(isPermitted(this.user, "admin")) ||
+              isPermitted(this.user, "owner")) {
+              return {
+                statusCode: 401,
+                status: "fail",
+                message: "you do not have permission to delete a record"
+              };
+            }
+            if (isPermitted(this.user, "admin") ||
+              isPermitted(this.user, "owner")) {
+              if (Collection._name === "Products") {
+                const item = Collection.findOne(this.urlParams.id);
+                item.isDeleted = true;
+                const updatedCollection = Collection.upsert({ _id: this.urlParams.id }, {
+                  $set: item
+                });
+                return {
+                  statusCode: 204,
+                  status: "success",
+                  message: "Product is archived",
+                  data: updatedCollection
+                };
+              }
+              const updatedCollection = Collection.remove({_id: this.urlParams.id });
+              return {
+                statusCode: 204,
+                status: "success",
+                message: "Collection is deleted",
+                data: updatedCollection
+              };
+            }
+          }
+        }
+
+        // patch: {
+        //   // TODO: write the patch action() method
+        // }
```

Comment:
Remove commented (dead) code.

Reply:
@johadi10 address this and let me know.

Reply:
@mradeybee Please review. I have removed the dead code